test/types/lean.test.ts

```diff
@@ -29,6 +29,7 @@ function gh10345() {
     expectError(leanDoc.id = 43);
 
     const doc2 = await UserModel.findOne().orFail().lean();
+    if (!doc2) return;
```

Comment:
This change is unfortunately a blocker. You should never need `if (!doc2)` if you have `orFail()`.